apis/policies/v1/clusteradmissionpolicy_types.go

```diff
@@ -151,7 +151,24 @@ func (r *ClusterAdmissionPolicy) GetRules() []admissionregistrationv1.RuleWithOp
 	return r.Spec.Rules
 }
 
-func (r *ClusterAdmissionPolicy) GetNamespaceSelector() *metav1.LabelSelector {
+func (r *ClusterAdmissionPolicy) GetNamespaceSelector(deploymentNamespace string) *metav1.LabelSelector {
+	// exclude namespace where kubewarden was deployed
+	if r.Spec.NamespaceSelector != nil {
+		r.Spec.NamespaceSelector.MatchExpressions = append(r.Spec.NamespaceSelector.MatchExpressions, metav1.LabelSelectorRequirement{
```

Comment:
I would look at all the existing Namespace selectors, and ensure this one that you're about to add, doesn't exist yet

Reply:
Raul pointed out on the PR description that we are already updating the webhook with a deepEqual comparison [here](https://github.com/kubewarden/kubewarden-controller/blob/main/internal/pkg/admission/mutating-webhook.go#L50). So I think we are good to go.

Reply:
unit tests added here https://github.com/kubewarden/kubewarden-controller/pull/338/commits/1c12617050c0bc15600d6588b41241c96e706833

@flavio I think it is better to add a new `MatchExpression` excluding the kubewarden namespace. Looking to existing ones is not that easy because we also need to look to all the different operations. From k8s [docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements)

> All of the requirements, from both matchLabels and matchExpressions are ANDed together -- they must all be satisfied in order to match.

So it doesn't really matter is this namespace also exists in a different `MatchExpression`

Reply:
Thanks for the clarification